src/gui/widgets/Fader.cpp

```diff
@@ -315,154 +284,207 @@ void Fader::updateTextFloat()
 }
 
 
-inline int Fader::calculateDisplayPeak( float fPeak )
-{
-	int peak = static_cast<int>(m_back.height() - (fPeak / (m_fMaxPeak - m_fMinPeak)) * m_back.height());
-
-	return qMin(peak, m_back.height());
-}
-
-
 void Fader::paintEvent( QPaintEvent * ev)
 {
 	QPainter painter(this);
 
-	// Draw the background
-	painter.drawPixmap(ev->rect(), m_back, ev->rect());
-
 	// Draw the levels with peaks
-	if (getLevelsDisplayedInDBFS())
-	{
-		paintDBFSLevels(ev, painter);
-	}
-	else
-	{
-		paintLinearLevels(ev, painter);
-	}
+	// TODO LMMS Compressor, EQ and Delay still use linear displays...
+	paintLevels(ev, painter, !getLevelsDisplayedInDBFS());
 
 	// Draw the knob
-	painter.drawPixmap(0, knobPosY() - m_knob.height(), m_knob);
+	painter.drawPixmap((width() - m_knob.width()) / 2, knobPosY() - m_knob.height(), m_knob);
 }
 
-void Fader::paintDBFSLevels(QPaintEvent * ev, QPainter & painter)
+void Fader::paintLevels(QPaintEvent * ev, QPainter & painter, bool linear)
 {
-	int height = m_back.height();
-	int width = m_back.width() / 2;
-	int center = m_back.width() - width;
+	std::function<float(float value)> mapper = [this](float value) { return ampToDbfs(qMax<float>(0.0001, value)); };
+
+	if (linear)
+	{
+		mapper = [this](float value) { return value; };
+	}
 
-	float const maxDB(ampToDbfs(m_fMaxPeak));
-	float const minDB(ampToDbfs(m_fMinPeak));
+	float const mappedMinPeak(mapper(m_fMinPeak));
```

Comment:
With most of the variable types here, I'd rather `const` be on the left (`const float`) to be more in line with the majority of the codebase.

Reply:
Done with commit b3a1dd9e69d although I don't like this ambiguity in C++. If you always write them to the right then you can simply read statements from right to left to get an exact idea of what something is.
```
int const * foo;
```
`foo` is a (non-const) pointer to a constant integer value.
```
int * const bar;
```
`bar` is a const pointer to a non-const integer value.
```
int const * const baz;
```
`baz` is a const pointer to a const integer value.

Reply:
The C++ standard or the coding guidelines of LMMS? :wink:

Reply:
Our coding conventions of course. The standard doesn't define which order I'm pretty sure. 😂

Reply:
The standard usually goes along the line of "everybody can do however they please and here are 100 options". 😂

Would be nice to change this in LMMS' coding guidelines but I guess it would be a lot of work to adjust this everywhere in the code.

Reply:
It would be a lot of work, yes. We'd also have to put it up to a vote.

Reply:
I guess it might also potentially break several existing pull requests. So it's a bit of a question if it is really worth it (even though I like "const to the right" :wink:).